var/spack/repos/builtin/packages/lua/package.py

```diff
@@ -289,10 +283,31 @@ def install(self, spec, prefix):
                         os.symlink(src_path, dest_path)
 
     @run_after("install")
-    def link_pkg_config(self):
+    def generate_pkg_config(self):
         if self.spec.satisfies("+pcfile"):
```

Comment:
I've checked and `rpm` is the only package that explicitly depends on `lua+pcfile` and no package explicitly depends on `lua~pcfile`. As you said, we could remove the variant `pcfile` and always generate the pc file.